app/services/update_interviews_provider.rb

```diff
@@ -0,0 +1,43 @@
+class UpdateInterviewsProvider
+  include ImpersonationAuditHelper
+
+  attr_reader :auth, :interviews, :provider, :application_choice
+
+  def initialize(
+    actor:,
+    application_choice:,
+    provider:
+  )
+    @auth = ProviderAuthorisation.new(actor: actor)
+    @application_choice = application_choice
+    @provider = provider
+    @interviews = application_choice.interviews
+  end
+
+  def save!
+    auth.assert_can_set_up_interviews!(application_choice: application_choice,
+                                       course_option: application_choice.current_course_option)
+
+    interviews.upcoming.each do |interview|
+      next if provider_is_the_same?(interview)
+
+      interview.provider = provider
+
+      InterviewWorkflowConstraints.new(interview: interview).update!
+
+      interview_validations = InterviewValidations.new(interview: interview)
+
+      if interview_validations.valid?(:update)
+        audit(auth.actor) do
+          interview.save!
+        end
+      else
+        raise ValidationException, interview_validations.errors.map(&:message)
+      end
+    end
+  end
+
+  def provider_is_the_same?(interview)
+    interview.provider == provider || application_choice.current_accredited_provider == provider
```

Comment:
We need to check if the accredited provider is on the interview `interview.provider == application_choice.current_accredited_provider` as otherwise we end up updating the interview provider, also we need to fix/add a test